src/user-management/user-management.ts

```diff
@@ -294,6 +309,62 @@ export class UserManagement {
     return deserializeAuthenticationResponse(data);
   }
 
+  async authenticateWithSessionCookie({
+    sessionCookie,
+    cookiePassword = process.env.WORKOS_COOKIE_PASSWORD,
+  }: {
+    sessionCookie: string;
+    cookiePassword?: string;
+  }): Promise<
+    | AuthenticateWithSessionCookieSuccessResponse
+    | AuthenticateWithSessionCookieFailedResponse
+  > {
+    if (!cookiePassword) {
+      throw new Error('Cookie password is required');
+    }
+
+    if (!sessionCookie) {
+      return { authenticated: false };
+    }
+
+    const session = await unsealData<SessionCookieData>(sessionCookie, {
+      password: cookiePassword,
+    });
+
+    if (!session) {
+      return { authenticated: false };
+    }
+
+    if (!(await this.isValidJwt(session.accessToken))) {
+      return { authenticated: false };
```

Comment:
I feel like we should return some sort of error or reason in these scenarios. Based on the returned object it's impossible to know if authentication failed because there isn't a session or if the access token has expired. Perhaps something like 

```ts
return { 
  authenticated: false,
  reason: 'access token expired',
}
```

Reply:
Yeah, that's a great idea. Right now it's pretty opaque. Will add!